components/raftstore/src/store/read_queue.rs

```diff
@@ -234,7 +245,14 @@ impl ReadIndexQueue {
         }
         self.ready_cnt -= 1;
         self.handled_cnt += 1;
-        self.reads.pop_front()
+        let res = self
+            .reads
+            .pop_front()
+            .expect("read_queue is empty but ready_cnt > 0");
+        // If it's a follower it's necessary, otherwise the field
```

Comment:
How about using a flag to avoid false remove?

Reply:
You mean `is_leader` like `push_back`? But if the queue is on a leader `contexts` will must be empty. So does it make coe more complex?